File: 백준/Bronze/2167. 2차원 배열의 합/2차원 배열의 합.py
import sys
input = lambda:sys.stdin.readline().strip()

N, M = map(int, input().split())
arr = [list(map(int, input().split())) for _ in range(N)]
for _ in range(int(input())):
    i, j, x, y = map(int, input().split())
    total = 0
    for a in range(i-1, x):
        total += sum(arr[a][j-1:y])
    print(total)

File: 백준/Bronze/2167. 2차원 배열의 합/2차원 배열의 합.js
const fs = require('fs');
// const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt"
// let input = fs.readFileSync(filePath).toString().trim().split("\n");
let input = fs.readFileSync('/dev/stdin').toString().trim().split("\n");

const [N, M] = input[0].split(' ').map(Number);
const arr = []

for (let i = 1; i < N+1; i++) {
    const arr_tmp = input[i].split(' ').map(Number);
    arr.push(arr_tmp);   
}
const K = Number(input[N+1])

for (let idx = N+2; idx < K+N+2; idx++) {
    const [a, b, x, y] = input[idx].split(' ').map(Number);
    let sum = 0
    // console.log(info)
    for (let i = a-1; i < x; i++) {
        for (let j = b-1; j < y; j++) {
            sum += arr[i][j];
        }
    }
    console.log(sum)
}

